### PR TITLE
Don't automatically add archive indexes in STRICT mode

### PR DESCRIPTION
**Pull request by sbc100**
Also, don't misclassify empty archives as archives that need to have
an index added.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -477,8 +477,12 @@ def do_emscripten(infile, memfile):
 
 
 def ensure_archive_index(archive_file):
-  # Fastcomp linking works without archive indexes
-  if not shared.Settings.WASM_BACKEND:
+  # Fastcomp linking works without archive indexes.  In SRICT mode we expect you to use the
```
Comment (kripken) [inline emcc.py]: typo in STRICT

**Comment by sbc100** [comment]
If a native toolchain, and in STRICT mode you will see the `archive has no index; run ranlib to add one` error message, which clearly tells you what you need to need to do to add an index to the archive.

Nowhere do we say disable STRICT mode since that would almost never be the right thing to do.  The right thing to do is fix your build system to use the correct `ar` or `ranlib` command.

TBH, I kind of which we'd never added this auto-magical feature.

**Comment by sbc100** [comment]
But you are right a new setting does make more sense.  Adding one

**Comment by sbc100** [comment]
Done

**Pull request merged**